x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -698,6 +698,23 @@ class AgentPolicyService {
           }),
     };
 
+    // Only add permissions if output.type is "elasticsearch"
+    fullAgentPolicy.output_permissions = Object.keys(fullAgentPolicy.outputs).reduce<
+      NonNullable<FullAgentPolicy['output_permissions']>
+    >((permissions, outputName) => {
+      const output = fullAgentPolicy.outputs[outputName];
+      if (output && output.type === 'elasticsearch') {
+        permissions[outputName] = {};
+        permissions[outputName].fallback = [
+          {
+            names: ['logs-*', 'metrics-*', 'traces-*', '.logs-endpoint.diagnostic.collection-*'],
+            privileges: ['auto_configure', 'create_doc'],
```

Comment:
I think this depends on what we want to do with the `fleet_enroll` role. Will it always keep these permissions or will it also have permissions based on the integrations?

If it's the first case, I would leave it as it is. The purpose of these permissions blocks is different and it will change independently from the agent's permissions.

If it's the second case then maybe it is wise to deduplicate it.

Reply:
good point, let's treat `fleet_enroll` separately for now